### PR TITLE
Update snakeyaml to 1.27 using the plugin api

### PR DESCRIPTION
**Pull request by MRamonLeon**
Bump SnakeYaml to 1.27 using the Jenkins plugin instead of the library directly.
Depends on https://github.com/jenkinsci/google-kubernetes-engine-plugin/pull/138 to allow the CI run.

cc: @alecharp @batmat @rsandell @fqueiruga @Wadeck

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -208,10 +208,10 @@
...
     </dependency>
     <dependency>
-      <groupId>org.yaml</groupId>
-      <artifactId>snakeyaml</artifactId>
-      <version>1.19</version>
-    </dependency>
+      <groupId>io.jenkins.plugins</groupId>
+      <artifactId>snakeyaml-api</artifactId>
+      <version>1.27.0</version>
+      <type>hpi</type>
```
Comment (Wadeck) [inline pom.xml]:
```suggestion
```

No need for it, otherwise you have to update the doc: https://plugins.jenkins.io/snakeyaml-api/ ;)

**Review by Wadeck — APPROVED** [review]

**Comment by MRamonLeon** [comment]
The CI failure doesn't seem related to the changes:

```
[2021-01-14T15:30:12.746Z] Recording test results
[2021-01-14T15:30:12.782Z] No test report files were found. Configuration error?
No test report files were found. Configuration error?
```

**Review by alecharp — APPROVED** [review]

**Comment by batmat** [comment]
@MRamonLeon it's now conflicted, can you fix it?

**Comment by MRamonLeon** [comment]
I believe it's failing because the CI is not taking into account the Jenkinsfile to update the Jenkins we run in. Could anyone update it?
* @craigdbarber 
* @evandbrown 
* @stephenashank 
* @donmccasland 
* @tequilarista 
* @vishal84 

I've also filed a new PR with just this change, if it helps: https://github.com/jenkinsci/google-kubernetes-engine-plugin/pull/138

To know more about the Jenkinsfile change not allowed to me: https://support.cloudbees.com/hc/en-us/articles/227246367-Why-Jenkinsfile-changes-Are-Not-Reflected-in-PR-Build

Thanks!

**Comment by MRamonLeon** [comment]
I finally added the BOM to better manage the dependencies.

**Comment by alecharp** [comment]
@donmccasland please see this PR as well as it important for the general health of the plugin. Thanks.

**Comment by alecharp** [comment]
@MRamonLeon could you merge `develop` branch in the PR to try to solve the build? Thanks.

**Comment by MRamonLeon** [comment]
> @MRamonLeon could you merge `develop` branch in the PR to try to solve the build? Thanks.

Is it needed? I think the problem was the Jenkins version the CI was running in. Once changed with #138 all should be fine.

**Comment by donmccasland** [comment]
I just merged #138 and this one as well.

**Comment by MRamonLeon** [comment]
@donmccasland Do you have any date in mind to release it? Thank you!

**Comment by donmccasland** [comment]
Not until Friday.  I'll update Jenkinsfile some more, merge the latest
batch of dependabot PRs and then release.

**Pull request merged**